### PR TITLE
Add-dev-container

### PR DESCRIPTION
**Pull request by sarg3nt**
Add a mvp dev container.  Still needs a lot of work but will suffice for now.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    // If you will be working with Kubernetes you should have a configured kube config file in your .kube directory.
+    "source=${localEnv:HOME}/.kube,target=/home/vscode/.kube-localhost,type=bind",
+    "source=${localEnv:HOME}/.ssh,target=/home/vscode/.ssh-localhost,type=bind",
+    "source=${localEnv:HOME}/.config/k9s,target=/home/vscode/.config/k9s-localhost,type=bind",
+    "source=${localEnv:HOME}/.local/share/k9s,target=/home/vscode/.local/share/k9s-localhost,type=bind",
+    "source=/var/run/docker.sock,target=/var/run/docker-host.sock,type=bind"
+  ],
+  "containerEnv": {
+    ////////// Repo Specific Environment Variables //////////
+    "MY_VAR": "${localEnv:MY_VAR}",
```
Comment (aidanleuck) [inline .devcontainer/devcontainer.json:25]: Is MY_VAR supposed to be set?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        },
+        "go.useLanguageServer": true,
+        "go.testEnvVars": {
+          "CGO_ENABLED": "1"
+        },
+        "go.testFlags": [
+          "-v",
+          "-race"
+        ],
+        "go.testTimeout": "10s",
```
Comment (aidanleuck) [inline .devcontainer/devcontainer.json:88]: Might not be a bad idea to raise this from 10s, maybe 30s?

**Review by aidanleuck — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor changes

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (sarg3nt) [reply .devcontainer/devcontainer.json:25]: It's left in as an example to future users but should be commented out.  Fixed

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (sarg3nt) [reply .devcontainer/devcontainer.json:88]: Done

**Review by aidanleuck — APPROVED** [review]

**Pull request merged**